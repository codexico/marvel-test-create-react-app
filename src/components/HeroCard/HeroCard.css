.HeroCard {
  display: flex;

  flex-direction: column;
  flex-wrap: wrap;
  justify-content: center;

  margin: 0 20px 20px;
  padding: 10px;
}

.HeroCard-img {
  width: 240px;
  height: 240px;
  margin-bottom: 10px;

  border-bottom: 5px solid red;

  object-fit: cover;
}

.HeroCard-name {
  font-size: 20px;
  font-weight: bold;
}
