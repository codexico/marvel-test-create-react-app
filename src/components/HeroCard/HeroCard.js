import React from 'react';
import PropTypes from 'prop-types';

import './HeroCard.css';

function HeroCard({ name, thumbnail }) {
  return (
    <div className="HeroCard">
      <img
        className="HeroCard-img"
        src={`${thumbnail.path}.${thumbnail.extension}`}
        alt={name}
      />
      <p className="HeroCard-name">{name}</p>
    </div>
  );
}

HeroCard.propTypes = {
  name: PropTypes.string.isRequired,
  thumbnail: PropTypes.shape({
    path: PropTypes.string,
    extension: PropTypes.string.isRequired,
  }).isRequired,
};

export default HeroCard;
