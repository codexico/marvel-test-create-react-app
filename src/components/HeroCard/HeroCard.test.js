import React from 'react';
import { render } from '@testing-library/react';

import HeroCard from './HeroCard';

test('Component renders correctly', () => {
  const { asFragment } = render(
    <HeroCard
      key={1010354}
      name={'Adam Warlock'}
      thumbnail={{
        path: 'http://i.annihil.us/u/prod/marvel/i/mg/a/f0/5202887448860',
        extension: 'jpg',
      }}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
