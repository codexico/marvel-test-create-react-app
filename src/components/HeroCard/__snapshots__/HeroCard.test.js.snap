// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="HeroCard"
  >
    <img
      alt="Adam Warlock"
      class="HeroCard-img"
      src="http://i.annihil.us/u/prod/marvel/i/mg/a/f0/5202887448860.jpg"
    />
    <p
      class="HeroCard-name"
    >
      Adam Warlock
    </p>
  </div>
</DocumentFragment>
`;
